src/main/java/ch/njol/skript/SkriptEventHandler.java

```diff
@@ -137,13 +142,20 @@ private static void check(Event event, EventPriority priority) {
 
 		for (Trigger trigger : triggers) {
 			SkriptEvent triggerEvent = trigger.getEvent();
-			if (triggerEvent.getEventPriority() != priority || !triggerEvent.check(event))
+			if (triggerEvent.getEventPriority() != priority)
 				continue;
 
 			logTriggerStart(trigger);
 			Object timing = SkriptTimings.start(trigger.getDebugLabel());
-
-			trigger.execute(event);
+			if (!Bukkit.isPrimaryThread() && !trigger.getEvent().canExecuteAsynchronously()) {
```

Comment:
the callSync will do its own primary thread check - do you think it's worth removing that check here then?